From 2e50ca323e97881d24481737ae6d5a52936d9047 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Fri, 29 Mar 2013 09:44:41 +0800
Subject: ENGR00256543 mx6: Update equation for thermal sensor on i.MX6DL

Use universal equation and 25C's calibration data to
get thermal sensor's ratio on i.MX6DL.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 drivers/mxc/thermal/thermal.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/thermal/thermal.c b/drivers/mxc/thermal/thermal.c
index 21a87b7..4931ce1 100644
--- a/drivers/mxc/thermal/thermal.c
+++ b/drivers/mxc/thermal/thermal.c
@@ -883,13 +883,13 @@ static int anatop_thermal_counting_ratio(unsigned int fuse_data)
 	raw_hot = (fuse_data & 0xfff00) >> 8;
 	hot_temp = fuse_data & 0xff;
 
-	if (!calibration_valid && cpu_is_mx6q())
+	if (!calibration_valid && !cpu_is_mx6sl())
 		/*
 		 * The universal equation for thermal sensor
 		 * is slope = 0.4297157 - (0.0015976 * 25C fuse),
 		 * here we convert them to integer to make them
 		 * easy for counting, FACTOR1 is 15976,
-		 * FACTORs is 4297157. Our ratio = -100 * slope.
+		 * FACTOR2 is 4297157. Our ratio = -100 * slope.
 		 */
 		ratio = ((FACTOR1 * raw_25c - FACTOR2) + 50000) / 100000;
 	else
-- 
1.7.9.5

